syntax = "proto3";

package encoderservice;

service EncoderService  {
  rpc Encode(EncodeRequest) returns (EncodeResponse);
  rpc CountTokens(CountTokensRequest) returns (CountTokensResponse);
  rpc SplitText(SplitTextRequest) returns (SplitTextResponse);
}

message EncodeRequest {
  string text = 1;
//  string model_name = 2; // optional
}

message EncodeResponse {
  repeated float vector = 1;
//  string model_name = 2;
}

message CountTokensRequest {
  string text = 1;
//  string model_name = 2;
}

message CountTokensResponse {
  int32 token_count = 1;
//  string model_name = 2;
}

message SplitTextRequest {
  string text = 1;
  int32 chunk_size = 2;
  int32 overlap = 3;
}

message SplitTextResponse {
  repeated string fragments = 1;
}
